docs/broker.rst

```diff
@@ -88,6 +88,73 @@ its secret type updates to the secret object in the Kubernetes API will be autom
 a Pod's filesystem. Hpfeeds will spot those updates and process them immediately without needing a
 restart.
 
+Database authentication store
+--------------------------
+
+When starting the broker you can pass a database connection string. Auth requests are then checked against
+the selected Database in a table named auth_keys. 
+
+Supported Database drivers are:
+  - postrgesql
+  - mysql
+  - mysql compatable e.g. mariadb, aurora
+  - sqlite
+
+You may need to install the specific database driver for your selected database, more information can be 
+found at https://pypi.org/project/databases/
+
+.. code-block: bash
+
+  $ pip install databases[postgresql]
+  $ pip install databases[mysql]
+  $ pip install databases[sqlite]
+
+Using SQLite with this auth mechanism requires JSON support that can be found in SQLite version > 3.3 and Python3
+Previous versions of SQLite may be supported with the JSON1 SQLite extension.
+
+Any authentication can be included within the connection string
+For example:
+
+.. code-block:: bash
+
+    hpfeeds-broker -e tcp:port=20000 --exporter=0.0.0.0:9431 --auth="database+mysql://honeyswarm:honeyswarm@127.0.0.1/honeyswarm"
+
+
+.. code-block:: bash
+
+    hpfeeds-broker -e tcp:port=20000 --exporter=0.0.0.0:9431 --auth="database+postgresql://localhost/example"
+
+.. code-block:: bash
+
+    hpfeeds-broker -e tcp:port=20000 --exporter=0.0.0.0:9431 --auth="database+sqlite:///auth_keys.db"
+
+To add a new user
+
+.. code-block:: bash
+
+    mysql -u admin -p <password>
+    use database_name;
+    
+    mysql> INSERT INTO auth_keys (identifier, secret, publish, subscribe) VALUES ('testing', 'secretkey', '["channel1", "channel2"]', '["channel2"]');
+    Query OK, 1 row affected (0.00 sec)
+
+
+To Find all users
+
+.. code-block:: bash
+
+    mysql -u admin -p <password>
+    use database_name;
+
+    mysql> select * from auth_keys;
+    +----+------------+-----------+--------------------------+--------------+
+    | id | identifier | secret    | publish                  | subscribe    |
+    +----+------------+-----------+--------------------------+--------------+
+    |  1 | testing    | secretkey | ["channel1", "channel2"] | ["channel2"] |
+    +----+------------+-----------+--------------------------+--------------+
+    1 row in set (0.00 sec)
+
+
```

Comment:
Can you document the schema for these databases? Does the same schema work for all 3 backends?

Reply:
The same schema works for all 3 backends so easy to add the schema in to the docs.

Reply:
added "create table" examples. Is it worth adding a setup tables function to always create these tables if they don't exist or let the user manage the DB